pkg/chunked/compression_linux.go

```diff
@@ -210,14 +210,18 @@ func readZstdChunkedManifest(blobStream ImageSourceSeekable, tocDigest digest.Di
 
 	decodedTarSplit := []byte{}
 	if tarSplitChunk.Offset > 0 {
-		tarSplit, err := readBlob(tarSplitChunk.Length)
-		if err != nil {
-			return nil, nil, nil, 0, err
-		}
-
-		decodedTarSplit, err = decodeAndValidateBlob(tarSplit, tarSplitLengthUncompressed, toc.TarSplitDigest.String())
-		if err != nil {
-			return nil, nil, nil, 0, fmt.Errorf("validating and decompressing tar-split: %w", err)
+		tarSplitDigest := toc.TarSplitDigest.String()
+		// ignore the tar-split data if the digest was not specified
```

Comment:
We must actually _consume_ the data, otherwise the producer of `(parts, errs)` is going to block forever (and keep the HTTP connection occupied).

Reply:
thanks!

Opened a PR: https://github.com/containers/storage/pull/1942